### DISCUSSION TITLE
How to render TabBarIndicator in custom TabBar?

### DISCUSSION THREAD

**Discussion by gkankava** [react-native-community/react-native-tab-view#1328] [Q&A]
Hi, can anyone help me to do it?

I want to render TabBarIndicator  from 
`import { TabView, TabBarIndicator } from "react-native-tab-view";`
inside custom tabbar.

```
  const renderTabBar = (props) => {
    return (
      <View>
        <TabBarIndicator {...props} />
      </View>
    );
  };
```

The problem is `<TabBarIndicator {...props} />` also needs getTabWidth function.

`<TabBarIndicator {...props} getTabWidth={ () => {} }  />`.  But i don't understand what to do now

**Comment by UnluckyLAD** [comment]
Exact issue I'm facing a the moment. Setting 'tabStyle' makes the indicator disappear. 

Example:
```
<TabBar
   {...props}
   **style={Styles.tabBar}**
   ...
/>
```
gives the following result:
![1](https://user-images.githubusercontent.com/40564700/164057427-3225937b-5e99-4e35-9f69-60e6ade553a5.png)

whereas the following:
```
<TabBar
   {...props}
   **tabStyle={Styles.tabBar}**
   ...
/>
```
returns the exact same UI but with no indicator.
![image](https://user-images.githubusercontent.com/40564700/164057758-e1f6ae7d-1d13-4a60-9043-a531fbd5e0b1.png)

I have a custom styled indicator which can be seen in the first screenshot, height of 3, and bg colour of white.
